package/src/components/MessageMenu/MessageUserReactions.tsx

```diff
@@ -0,0 +1,180 @@
+
+import {
+  MessagesContextValue,
+  useMessagesContext,
+} from '../../contexts/messagesContext/MessagesContext';
+import { useTheme } from '../../contexts/themeContext/ThemeContext';
+import { useTranslationContext } from '../../contexts/translationContext/TranslationContext';
+import { DefaultStreamChatGenerics, Reaction } from '../../types/types';
+import { ReactionData } from '../../utils/utils';
+import { MessageType } from '../MessageList/hooks/useMessageList';
+
+export type MessageUserReactionsProps<
+  StreamChatGenerics extends DefaultStreamChatGenerics = DefaultStreamChatGenerics,
+> = Partial<
+  Pick<
+    MessagesContextValue<StreamChatGenerics>,
+    'MessageUserReactionsAvatar' | 'MessageUserReactionsItem' | 'supportedReactions'
+  >
+> & {
+  /**
+   * The message object
+   */
+  message?: MessageType<StreamChatGenerics>;
+  /**
+   * An array of reactions
+   */
+  reactions?: Reaction[];
+};
+
+const sort: ReactionSortBase = {
+  created_at: -1,
+};
+
+export const MessageUserReactions = (props: MessageUserReactionsProps) => {
+  const {
+    message,
+    MessageUserReactionsAvatar: propMessageUserReactionsAvatar,
+    MessageUserReactionsItem: propMessageUserReactionsItem,
+    reactions: propReactions,
+    supportedReactions: propSupportedReactions,
+  } = props;
+  const reactionTypes = Object.keys(message?.reaction_groups ?? {});
+  const [selectedReaction, setSelectedReaction] = React.useState<string | undefined>(
+    reactionTypes[0],
+  );
+  const {
+    MessageUserReactionsAvatar: contextMessageUserReactionsAvatar,
+    MessageUserReactionsItem: contextMessageUserReactionsItem,
+    supportedReactions: contextSupportedReactions,
+  } = useMessagesContext();
+  const supportedReactions = propSupportedReactions ?? contextSupportedReactions;
+  const MessageUserReactionsAvatar =
+    propMessageUserReactionsAvatar ?? contextMessageUserReactionsAvatar;
+  const MessageUserReactionsItem = propMessageUserReactionsItem ?? contextMessageUserReactionsItem;
+
+  const messageReactions = reactionTypes.reduce<ReactionData[]>((acc, reaction) => {
+    const reactionData = supportedReactions?.find(
+      (supportedReaction) => supportedReaction.type === reaction,
+    );
+    if (reactionData) {
+      acc.push(reactionData);
+    }
+    return acc;
+  }, []);
+
+  const {
+    loading,
+    loadNextPage,
+    reactions: fetchedReactions,
+  } = useFetchReactions({
+    messageId: message?.id,
+    reactionType: selectedReaction,
+    sort,
+  });
+
+  const {
+    theme: {
+      messageMenu: {
+        userReactions: {
+          container,
+          flatlistColumnContainer,
+          flatlistContainer,
+          reactionSelectorContainer,
+          reactionsText,
+        },
+      },
+    },
+  } = useTheme();
+  const { t } = useTranslationContext();
+
+  const reactions = useMemo(
+    () =>
+      propReactions ||
+      (fetchedReactions.map((reaction) => ({
+        id: reaction.user?.id,
```

Comment:
Isn't `alignment` missing from here ? It's `required` according to the type.

Reply:
Ah no, the type of `Reaction` is simplified for better:

```
export type Reaction = {
  id: string;
  name: string;
  type: string;
  image?: string;
};
```

Reply:
Ah, okay - I think I was looking at the wrong thing. Was wondering how TS hasn't complained yet :D